packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -41,7 +41,15 @@ export abstract class AbstractResourcePreferenceProvider extends PreferenceProvi
             this._ready.resolve();
             return;
         }
-        this.resource = this.resourceProvider(uri);
+
+        // Determine if there are workspace preferences, else we skip it
+        try {
+            this.resource = this.resourceProvider(uri);
+            await this.resource;
+        } catch (error) {
+            this._ready.resolve();
+            return;
```

Comment:
the `init()` function ends here, and
```ts
resource.onDidChangeContents(content => this.readPreferences())
```
is not run, if an error is caught.

So I am curious if the `settings.json` would be read, if the file does not exist in the first place, and gets added by the user after theia finishes loading.